sopel/tests/pytest_plugin.py

```diff
@@ -5,11 +5,163 @@
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+import re
+import sys
+
+import py
 import pytest
 
+from sopel import bot, loader, plugins, trigger
 from .factories import BotFactory, ConfigFactory, IRCFactory, TriggerFactory, UserFactory
 
 
+TEMPLATE_TEST_CONFIG = """
+[core]
+nick = {name}
+owner = {owner}
+admin = {admin}
+"""
+
+
+def get_disable_setup():
+    """Generate a pytest fixture to setup the plugin before running its tests.
+
+    When using ``@example`` for a plugin callable with an expected output,
+    pytest will be used to run it as a test. In order to work, this fixture
+    must be added to the plugin to set up the plugin before running the test.
+    """
+    @pytest.fixture(autouse=True)
+    def disable_setup(request, monkeypatch):
+        setup = getattr(request.module, "setup", None)
+        isfixture = hasattr(setup, "_pytestfixturefunction")
+        if setup is not None and not isfixture and py.builtin.callable(setup):
```

Comment:
Point of interest: `py` is in maintenance mode. Would it be too much trouble to replace its single use here with something supported, or ideally something that needs only stdlib?

I guess `pytest` itself depends on `py`, but that's not guaranteed to remain the case. If Sopel's own code is going to import `py`, it should be in Sopel's own requirements list.

Reply:
I suggest that we do that in another PR. I really don't want to change too much in one PR, given that here I'm just moving code around without changing any behaviour.

If you don't mind.

Reply:
Wait. I just realized... we are using `py` just for... `callable`??? What the...

OK nevermind, I'll remove that. It's stupid.